contracts/Depository.sol

```diff
@@ -326,16 +388,34 @@ contract Depository is IErrorsTokenomics {
 
         // Decrease the supply for the amount of payout
         supply -= payout;
+        // Adjust payout and set supply to zero if supply drops below the min defined value
+        if (supply < minOLASLeftoverAmount) {
+            payout += supply;
+            supply = 0;
+        }
```

Comment:
If the product supply drops below a specified value, it's remainder is given to the bonder, and the product is instantly closed.